### DISCUSSION TITLE
Running Webots in GitLab-CI Pipeline: `--log-performance`

### DISCUSSION THREAD

**Discussion by ThorbenCarl** [cyberbotics/webots#6166] [General]
Hello together,
we have built a simulation for our robot and now we have defined test cases and test them in GitLab CI/CD pipeline. 
Works also everything best, however I would like to use the function `--log-performance`.
Unfortunately it doesn't work for me or I can't find the file. Must there already be an empty file or is it created automatically? Which file format must the file have .txt .log ...? Or is it a problem that I kill Webots?

[GitLabYML.txt](https://github.com/cyberbotics/webots/files/11629963/GitLabYML.txt)
line 29: starting Webots
line 39: looking for log-file

[Job_log.txt](https://github.com/cyberbotics/webots/files/11629961/Job_log.txt)
line 169: no log-file available

Am I missing something or what am I doing wrong? 

(I know that there is a supervisor, until now I did not have the time to deal with it further. But it is definitely on the to-do list!)

**Comment by ygoumaz** [comment]
There are two potential issues that could be causing the problem:

1. Make sure you are providing the correct file path for the `log-performance `option. If you are using a relative path, keep in mind that it will be relative to the Webots installation folder. If the sim folder does not exist in the Webots installation folder, the log file will not be created.

2.  When closing Webots, it's important to ensure a clean shutdown process. Killing the Webots process abruptly may prevent the creation of the log file. Since you are running Webots headless with xvfb-run, it is recommended to use a supervisor to terminate the simulation. This will allow Webots to properly close and finalize any necessary operations, including the creation of the log file.
